### ISSUE TITLE
Idea: also support tagged template strings

### ISSUE THREAD

**Issue by stubailo** [apollostack/vscode-graphql#1]
Personally I think I will prefer to write my queries in the same file as my component, so would it be easy to add the same feature inside tagged template strings?

Like so:

```
query: gql`
  { ... }
`
```

I'm going to build a similar thing for Atom - let me know if you have some implementation ideas, I think we can reuse most of GraphiQL!

**Comment by stubailo** [comment]
Oh, just saw you already had this in the README. Personally, I think it would be great to open issues for all of them, so that we can discuss the possible implementation!

That way we can collaborate since I'm working on very similar stuff, for example this: https://github.com/apollostack/eslint-plugin-graphql

I also want to build the same things for Atom.

**Comment by mquandalle** [comment]
Yes, :+1: for that. I haven’t find the way to support ES6 template string, so I asked on StackOverflow: http://stackoverflow.com/questions/36795202/vscode-extension-language-inlining-with-es6-template-strings, let’s see how it goes.

**Comment by mquandalle** [comment]
For the record, leaving a ref to the VSCode issue of embedded languages in ES6 template strings: https://github.com/Microsoft/vscode/issues/5961

**Comment by stubailo** [comment]
@mquandalle thanks for opening this! I agree that there are a lot of examples where it's relevant, like `<script>` tags inside HTML, for example.

**Comment by rohmanhm** [comment]
Now, it's possible.
Look how `styled-component` do this https://github.com/styled-components/vscode-styled-components

**Comment by stubailo** [comment]
I think this package is more actively supported: https://github.com/kumarharsh/graphql-for-vscode

Perhaps, we should recommend people use that instead.

**Comment by Billy-** [comment]
FWIW, while that plugin looks like it's more complete, it has several dependencies that have to be manually installed. That's enough to put me off, but as there was nothing else that did what I wanted (validating ES6 template strings) I gave it a try. And failed. Couldn't get it working.

**Comment by jaydenseric** [comment]
I'm migrating from Atom to VS Code and finding the state of GraphQL extensions is a bit lacking.

I don't want to use [kumarharsh/graphql-for-vscode](https://github.com/kumarharsh/graphql-for-vscode) for these main reasons:

* [It requires a `.gqlconfig` file instead of using `.graphqlconfig`](https://github.com/kumarharsh/graphql-for-vscode/issues/40#issuecomment-343800065).
* [It does it's own linting, which is better handled separately by `eslint-plugin-graphql`](https://github.com/kumarharsh/graphql-for-vscode/issues/30).
* [It's complicated to setup](https://github.com/kumarharsh/graphql-for-vscode/issues/42).

There is now a way to support language smarts within template strings: [Microsoft/typescript-template-language-service-decorator](https://github.com/Microsoft/typescript-template-language-service-decorator). Maybe this extension could be resurrected and brought up to speed?
